### PR TITLE
Track change count and last update for stack links and dps

### PR DESCRIPTION
**Pull request by anurag6**

### CODE REVIEW

File: forch/faucet_state_collector.py [hunk 1]
```diff
@@ -622,6 +630,8 @@ def process_stack_topo_change(self, timestamp, stack_root, graph, dps):
                 change = True
             if t_state.get(TOPOLOGY_GRAPH) != graph:
                 t_state[TOPOLOGY_GRAPH] = graph
+                t_state[GRAPH_CHANGE_COUNT] = t_state.setdefault(GRAPH_CHANGE_COUNT, 0) + 1
```
Comment (grafnu) [inline forch/faucet_state_collector.py]: setdefault is unnecessary (just use get)

File: forch/faucet_state_collector.py [hunk 2]
```diff
@@ -56,6 +56,8 @@ def wrapped(self, *args, **kwargs):
 TOPOLOGY_DPS = "dps"
 TOPOLOGY_CHANGE_COUNT = "dataplane_state_change_count"
 TOPOLOGY_LAST_CHANGE = "dataplane_state_last_change"
+GRAPH_CHANGE_COUNT = "links_change_count"
```
Comment (grafnu) [inline forch/faucet_state_collector.py]: graph -> links?  A graph includes both links and nodes!

File: forch/faucet_state_collector.py [hunk 3]
```diff
@@ -259,17 +261,22 @@ def _get_switch_map(self):
...
+                change_count += switch_state.get(SW_STATE_CHANGE_COUNT, 0)
+                last_change = max(last_change, switch_state.get(SW_STATE_LAST_CHANGE, ''))
                 if not current_state:
                     switch_map[switch][SW_STATE] = None
                 elif current_state == SWITCH_CONNECTED:
                     switch_map[switch][SW_STATE] = constants.STATE_ACTIVE
                 else:
                     switch_map[switch][SW_STATE] = constants.STATE_DOWN
-
+            switch_map['switches_change_count'] = change_count
```
Comment (grafnu) [inline forch/faucet_state_collector.py]: somewhere around line 143 these counts (both switches and links) should be rolled-up into the overall dataplane change count (and last change).

**Comment by anurag6** [comment]
@grafnu PTAL

File: forch/faucet_state_collector.py [hunk 4]
```diff
@@ -148,33 +152,48 @@ def get_dataplane_state(self):
...
+        dplane_state['dataplane_state_change_count'] = change_count
+        dplane_state['dataplane_state_last_change'] = last_change
+
         return dplane_state
 
     def _get_broken_switches(self, dplane_state):
         broken_sw = []
-        sw_map = dplane_state.get(TOPOLOGY_DP_MAP, {})
+        sw_map = dplane_state.get('switch', {}).get(TOPOLOGY_DP_MAP, {})
+        LOGGER.info('Anurag sw_map: %s', sw_map)
```
Comment (grafnu) [inline forch/faucet_state_collector.py]: debugging left over

File: forch/faucet_state_collector.py [hunk 5]
```diff
@@ -104,29 +106,30 @@ def wrapped(self, *args, **kwargs):
     def get_dataplane_summary(self):
         """Get summary of dataplane"""
         dplane_state = self._get_dataplane_state()
-        state, detail, count, last = self._get_dataplane_detail(dplane_state)
+        state = dplane_state.get('dataplane_state')
```
Comment (grafnu) [inline forch/faucet_state_collector.py]: these can all be inlined -- much shorter

File: forch/faucet_state_collector.py (hunk 4 above)
Comment (anurag6) [reply forch/faucet_state_collector.py]: sorry about that.

File: forch/faucet_state_collector.py (hunk 5 above)
Comment (anurag6) [reply forch/faucet_state_collector.py]: inlined?

**Pull request merged**